5.OAM_KPT_Demo/README.md

```diff
@@ -0,0 +1,336 @@
+# Open Application Model with KPT
+
+This sample will show you how to use KPT to manage OAM Apps.
+
+## Relationship between KPT and OAM
+
+See the picture below:
+
+![img](./resource/kpt_oam.png)
+
+KPT is the manipulator of data while OAM is the format of data. With the help of KPT, we could easily manipulate App YAML files in OAM abstraction.
+
+## Pre-requisites
+
+Install OAM runtime by following the [installation guides](https://github.com/oam-dev/crossplane-oam-sample#installation).
+
+## Building OAM App Repository with KPT
+
+With the help of KPT, we could directly use GitHub Repo as App Repository without organizing apps in any fixed format.
+
+### Release your OAM apps to App Repository
+
+So release your OAM app only need two steps.
+
+1. Create/Fork a github repo.
+
+2. Commit and push your app.
+    ```
+    git add sampleapp/
+    git commit -m "my sampleapp commit"
+    git remote add origin git@github.com:<your-account>/<your-app-repo>.git
+    git push -u origin master
+    ```
+
+### Fetch OAM app from remote Repository
+
+Assuming you have already built an OAM App repository by GitHub.
+
+Using our [example repository](https://github.com/oam-dev/sample/tree/master/5.OAM_KPT_Demo/repository/) for this demo.
+
+You could fetch OAM app from remote Repository using [kpt pkg get](https://googlecontainertools.github.io/kpt/reference/pkg/get/).
+
+#### Command
+
+```
+kpt pkg get https://github.com/oam-dev/samples.git/5.OAM_KPT_Demo/repository/sampleapp sampleapp
+```
+
+#### Output
+
+```
+fetching package /5.OAM_KPT_Demo/repository/sampleapp from https://github.com/oam-dev/samples to sampleapp
+```
+
+```
+➜  kpt tree sampleapp
+sampleapp
+├── Kptfile
+├── appconfig.yaml
+├── component.yaml
+├── traitdef.yaml
+└── workloaddef.yaml
+
+0 directories, 5 files
+```
+
+### Install sample app
+
+```
+$ kubectl apply -f sampleapp/
+applicationconfiguration.core.oam.dev/example-appconfig created
+component.core.oam.dev/example-component created
+traitdefinition.core.oam.dev/manualscalertraits.core.oam.dev created
+workloaddefinition.core.oam.dev/containerizedworkloads.core.oam.dev created
+```
```

Comment:
Should explain what are the above resources. e.g.

- `Component` is the description of your application.
- `ApplicationConfiguration` is the description of operational policies for your `Component`.
- `TraitDefinition` and `WorkloadDefinition` are internal CRDs for OAM plugin to use.